lib/retry.js

```diff
@@ -116,6 +136,12 @@ function retryOn(attempt, error, options) {
                 return false;
             }
         }
+
+        // if function provided, determine whether to retry given the HttpResponseError
+        if (options.retryOnHttpResponseError && (error instanceof HttpResponseError)) {
```

Comment:
should we also check if `retryOnHttpResponseError` is a function and not sent in as `true`, otherwise maybe add a failing test when `retryOnHttpResponseError = true`

Reply:
Good catch, I added a check to confirm `retryOnHttpResponseError` is a `Function`, and a unit test to confirm non-function values do not result in a retry.

```
if (options.retryOnHttpResponseError && options.retryOnHttpResponseError instanceof Function && error instanceof HttpResponseError) { }
```